### PR TITLE
Show friendly 500 error page to users and developers

### LINKED ISSUE #24104: Nested 500 page inside a page
### Description

The position of 500 error might be incorrect if an error occurs inside a nested template. In the example below, the error occurs in line 694 of [`comments.tmpl`](https://github.com/go-gitea/gitea/blob/2b749af50508a28c7088e7d76132c27f96a45789/templates/repo/issue/view_content/comments.tmpl#L694), and this 500 error page was nested around this place where the error occurs.

https://user-images.githubusercontent.com/17645053/231715112-e151b595-d5e2-4344-a059-64671e359d99.mov

Steps to reproduce:

1. Create a team under an organization with the following settings

截屏2023-04-13 17 27 11

2. Open a pull request and assign the team as the reviewer.

截屏2023-04-13 17 32 27

3. Delete this team.
4. Go back to the pull request where this team was formerly assigned.

And the following JS error occurs because when the 500 error page is nested, there will be two `head_script.tmpl` with `webcomponents.js` imported twice. This error will be fixed if 500 page is not nested. 

截屏2023-04-13 17 47 40

### Gitea Version

latest

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

On macOS:
```
TAGS="bindata sqlite sqlite_unlock_notify" make build
./gitea
```

### Database

None

### PR DESCRIPTION
**Pull request by wxiaoguang**
Close #24104

This also introduces many tests to cover many complex error handling functions.

### Before

The details are never shown in production.

<details>

image

</details>

### After

The details could be shown to site admin users. It is safe.

image

### CODE REVIEW

File: templates/devtest/tmplerr-sub.tmpl [hunk 1]
```diff
@@ -0,0 +1,3 @@
+sub template triggers an executing error
```
Comment (silverwind) [inline templates/devtest/tmplerr-sub.tmpl:1]: `templates/test`? They aren't really used anywhere that I see but I guess we can start test-only templates.

Comment (wxiaoguang) [reply templates/devtest/tmplerr-sub.tmpl:1]: It's only enabled in dev mode, it helps development and testing, so what's why it's called `devtest`, and this word is unique in the codebase, people can easily grep it out.


image

Comment (silverwind) [reply templates/devtest/tmplerr-sub.tmpl:1]: I see, it's already in use, so lets keep it.

**Review by silverwind — APPROVED** [review]

**Comment by HesterG** [comment]
Tested and the issue is fixed, so LGTM

**Pull request merged**